### ISSUE TITLE
Volume control?

### ISSUE THREAD

**Issue by bgottsch** [KraigM/homebridge-harmonyhub#43]
Wouldn't it be possible to control volume with the new Speaker service introduced with iOS 10 beta?

**Comment by jer78** [comment]
Curious where you saw there's a new speaker service??

**Comment by bgottsch** [comment]
@jer78 In iOS 10 they added the ability to add doorbells and cameras, which made them add a speaker and microphone service. Since it's still in beta, only developers can access and the details are covered under an NDA contract. Though when it becomes public, this service might be able to add the volume ability to Harmony Hub. Also, I don't know if Siri will support it. Still, volume control with HomeKit would be awesome.

**Comment by jer78** [comment]
Ah ok. Yes. I was thinking you meant there was a new service specifically for a speaker in the latest beta. I guess it's wishful thinking that Apple would support their AirPlay speakers at some point in HomeKit.

**Comment by bgottsch** [comment]
Couldn't agree more.

Anyway, this may be a work around to add the ability. Probably when homebridge adds support for the service, we would be able to explore it more.

**Comment by jer78** [comment]
Agreed. It would be nice to have. Right now I use the AppleScript File Lightbulb plugin for my AirPlay speakers but at times confuses them with lights when I say "turn off all lights". Annoying work around to get speakers with volume control.

**Comment by akarmakar** [comment]
not sure this will helpfull here is the repo used to control volume with alexa 
https://github.com/sushilks/alexaHarmonyApp

**Comment by KraigM** [comment]
Im definitely interested in this too 😄.  Is anyone working on this yet?

**Comment by JangoBritt** [comment]
oh! I so hope this works!

If you've got a module now for harmony command functionality... does that mean that potentially it could use be used for other device commands? like Roomba: Clean and Roomba: Dock, without switching the activities on?

**Comment by arndth** [comment]
Should it work already? Or how could I volume up or maybe mute my TV with siri?

**Comment by larspetter** [comment]
Volume is working with the Sonos Homebridge plugin using Eve app. So guess if the Harmony API supports it it's possible?

**Comment by David-Noble-at-work** [comment]
KraigM, What's the status of this feature work? Are you looking for help, either development or test?

**Comment by rfiorentino1** [comment]
Do I need to download this commit and change a file I have? Or is it merged to the master yet? Currently, after just installing this plugin through the terminal Siri will not control volume of my TV through the Harmony Hub. Thoughts/suggestions?

**Comment by vonny3** [comment]
Also curious on the status of this - would be a great addition to have included.

**Comment by aguaman15** [comment]
If you want to control your device's volume with Siri, I've figured out a simple workaround that requires no additional code or software that you don't already have installed (i.e., Homebridge, the Harmony app, HomeKit, and this EXCELLENT HarmonyHub plugin). 

The main problem is that the Harmony won’t activate an activity if it is already on. We can create an activity, for example, that increases or decreases the volume by 1, but we can't activate that same activity again to raise the volume again. We need some way to automatically turn off the activity (without turning off the device) so we can activate it again to raise the volume again. To work around that limitation, I created a toggle using the Harmony app, the HarmonyHub plugin, and HomeKit rules. 

First, I set the Device's power settings to switch between activities without powering off by using the Harmony app. Select the Devices tab --> Edit Devices --> your device --> Power Settings --> select Keep Device Always ON but Switch Off when OFF Button is Pressed, then select the right arrow at the top, next select Using Two Different Buttons, and select the right arrow at the top again twice. Lastly press the "X" in the upper right hand corner. 

Then I created a Harmony activity called STEREO NULL which is really just a placeholder and meant to do absolutely nothing other than power on the receiver. Make sure there is no input set for the activity.  Afterward, I created 2 other activities called VOLUP and VOLDOWN (so the names didn’t conflict with the Siri names I wanted), also with no input settings. Next I edited the VolUp activity and added 5 VolumeUp command steps to the Start Sequence (add as many or few as you like), and did the same thing for VolDown. Then restarted Homebridge.

The next step was to setup the Siri names I wanted to use. I created 3 scenes in HomeKit that switched on each of the activities above. I named the scenes: Stereo Volume Up, Stereo Volume Down, and Stereo Null.   

The last step was to create 2 HomeKit rules. One for volume up and one for volume down. The rules were:

Trigger - VOLUP=on
Condition - STEREO NULL=off
Action - STEREO NULL=on

Whenever the VOLUP activity is triggered, the STEREO NULL activity is triggered right afterward, de-toggling the VOLUP activity so that you can activate it again right away. So now it toggles between VOLUP and STEREO NULL and you can raise the volume as much as you like. Then I did the same thing for volume down. 😊

That’s it! I don’t know if it matters, but I used the EVE app to do all this.

**Comment by m4d-maNu** [comment]
works the Volume Control in the alpha Version? Because I do not see any in the Home App.
